xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-slash/plugin.js

```diff
@@ -256,6 +256,12 @@
       return withStrictHTMLEncoding(() => {
         return originalAutoCompletePrototype.getHtmlToInsert.apply(this, args);
       });
+    },
+    getTextWatcher: function(textTestCallback) {
+      // Update the drop-down on click for shortcut triggering Quick Actions, see XWIKI-21062.
+      const textWatcher = new CKEDITOR.plugins.textWatcher(this.editor, textTestCallback, this.throttle);
+      this.editor.on("afterInsertHtml", function () {textWatcher.check(false);});
```

Comment:
The comment above is not very clear in explaining **why** we need to listen to ``afterInsertHtml``. AFAIU (based on https://github.com/ckeditor/ckeditor4/blob/4a1fb11f44b5b34f4834472d5717d03bfb2f9915/plugins/textwatcher/plugin.js#L195 ) the reason is that currently the text watcher listens mostly to keyboard events and command execution. When we use the Enter key to trigger a quick action the text watcher catches that keyboard event and if the quick action inserts some text then it reacts to that text (e.g. identifies that the inserted text matches the link shortcut). The same happens if the quick action triggers a command: even if we use the mouse click to trigger the quick action, the fact that the action triggers a command it enough to trigger the text watcher. The problem is when the quick action doesn't use a command and is activated with a click. For this we need to trigger the text watcher ourselves, after the HTML is inserted.